crates/host/src/policy.rs

```diff
@@ -1,26 +1,72 @@
-use core::time::Duration;
-
 use std::collections::{BTreeMap, HashMap};
 use std::hash::Hash;
-use std::sync::Arc;
 use std::time::{SystemTime, UNIX_EPOCH};
 
-use anyhow::Context;
-use futures::{
-    stream::{AbortHandle, Abortable},
-    StreamExt,
-};
 use serde::{Deserialize, Serialize};
-use tokio::spawn;
-use tokio::sync::RwLock;
-use tracing::{debug, error, instrument, trace, warn};
-use ulid::Ulid;
 use uuid::Uuid;
 use wascap::jwt;
 
 // NOTE: All requests will be v1 until the schema changes, at which point we can change the version
 // per-request type
-const POLICY_TYPE_VERSION: &str = "v1";
+pub(crate) const POLICY_TYPE_VERSION: &str = "v1";
+
+/// A trait for evaluating policy decisions
+#[async_trait::async_trait]
+pub trait PolicyManager: Send + Sync {
+    /// Evaluate whether a component may be started
+    async fn evaluate_start_component(
+        &self,
+        _component_id: &str,
+        _image_ref: &str,
+        _max_instances: u32,
+        _annotations: &BTreeMap<String, String>,
+        _claims: Option<&jwt::Claims<jwt::Component>>,
+    ) -> anyhow::Result<Response> {
+        Ok(Response {
+            request_id: Uuid::new_v4().to_string(),
```

Comment:
Nit: This means that any request with the default policy manager will still be spending cycles generating a uuid and allocating a string. We should probably do what NATS does and just return an empty string at least. We should also find a way to get rid of the string allocation in the future as well, but definitely not here